smarts/core/mission_planner.py

```diff
@@ -202,44 +202,48 @@ def _edge_ids(self, pose: Pose, lane_id: str = None):
         return edge_ids
 
     def cut_in_waypoints(self, sim, pose: Pose, vehicle):
-        radius = self._mission.task.trigger_radius
+
+        aggressiveness = 5
         neighborhood_vehicles = sim.neighborhood_vehicles_around_vehicle(
-            vehicle=vehicle, radius=radius
+            vehicle=vehicle, radius=80
         )
 
         position = pose.position[:2]
         lane = self._road_network.nearest_lane(position)
 
-        if neighborhood_vehicles:
-            nei_vehicle = neighborhood_vehicles[0]
+        if not neighborhood_vehicles or sim.elapsed_sim_time < 3:
+            return []
 
-            target_position = nei_vehicle.pose.position[:2]
-            target_lane = self._road_network.nearest_lane(target_position)
+        target_vehicle = neighborhood_vehicles[0]
+        target_position = target_vehicle.pose.position[:2]
+        target_lane = self._road_network.nearest_lane(target_position)
 
-            offset = self._road_network.offset_into_lane(lane, position)
-            target_offset = self._road_network.offset_into_lane(
-                target_lane, target_position
-            )
+        offset = self._road_network.offset_into_lane(lane, position)
+        target_offset = self._road_network.offset_into_lane(
+            target_lane, target_position
+        )
 
-            if offset < target_offset + 5 and not self._task_is_triggered:
-                return []
+        cut_in_offset = np.clip(20 - aggressiveness, 10, 20)
+        if (
+            abs(offset - (cut_in_offset + target_offset)) > 1
+            and lane.getID() != target_lane.getID()
+            and self._task_is_triggered is False
+        ):
             nei_wps = self._waypoints.waypoint_paths_on_lane_at(
-                target_position, target_lane.getID(), 60
+                position, lane.getID(), 60
+            )
+            speed_limit = np.clip(
+                (target_vehicle.speed + 5)
+                - 5.2 * (offset - (cut_in_offset + target_offset)),
+                np.clip(target_vehicle.speed - 5, 2.5, 30),
```

Comment:
`np.clip(..., target_vehicle.speed * 0.8, target_vehicle.speed * 1.2)`

Reply:
Thanks, I used `np.clip(..., target_vehicle.speed * 0.6, target_vehicle.speed * 1.4)` to add extra flexibility. Please let me know If you want me to change it.

Reply:
I still feel that 0.8, 1.2 is more reasonable/realistic because once the task is triggered, the cut-in speed is 1.15 times of the `target_vehicle.speed`.

Reply:
Sure, I change it back.